tables/basic-table/step9.md

```diff
@@ -0,0 +1,59 @@
+In this step, we'll find and fix a bug that you may or may not have already noticed in the current implementation, and then properly mark our callbaks that use state properties as react "callback hooks" so that they work as expected within the react data change detection ecosystem that is hooks.
+
+## Task
+
+1) Set the per per dropdown to "2 per page" and navigate to the very last page using the pagination controls, now select "3 per page" from the dropdown and notice our rows are gone!
+
+2) Locate the `handlePerPageSelect` handler function and replace the call to `updateList(currentPage);` with the following logic.
+
+<pre class="file" data-target="clipboard">
+    // when a user reaches end of pagination, and sets numPerPage to a value
+    // that requires more data than we have available, ensure the rows are set according
+    // to the previous page's index (currentPage - 1)
+    if (value > numPerPage && currentPage === defaultRows.length / numPerPage) {
+      updateList(currentPage - 1);
+    } else {
+      updateList(currentPage);
+    }
+</pre>
+
+3) Locate the dependency array for the useEffect hook you created earlier (`[numPerPage]`). Register the handlePerPageSelect function as a dependency of the function, right alongside of numPerPage. So changing `[numPerPage]` to `[numPerPage, handlePerPageSelect]`. This tells React to keep these values in sync.
+
+4) Now convert handlePerPageSelect to a callback hook by stuffing the function signature inside a call to React.useCallback(), and specify an array of dependencies for this callback hook like you did for handlePerPageSelect previously. Omitting the specifics, it should generally look like the following, where FN is your callback function and DEPS is any state properties or other callback functions (that themselves use state properties) this callback requires
+
+<pre class="file">
+const handlePerPageSelect = React.useCallback(<FN>, [<DEPS>]);
+</pre>
```

Comment:
I got confused by this and copied the code. Perhaps we should skip showing the function signature? That is, considering we provide the code just below.

Reply:
Good call, I moved this to being an inline snippet instead of a code block below, I think that should help make it obvious we're just showing the general structure without indicating it should be copied directly.

Reply:
Still a problem with the formatting, but I can see the  FN and DEPS now

<img width="548" alt="Screen Shot 2019-10-11 at 9 41 15 AM" src="https://user-images.githubusercontent.com/17481322/66656442-f9b9a780-ec0b-11e9-8bf0-95d95fcc9148.png">